radix-engine-interface/src/model/resource/non_fungible_id.rs

```diff
@@ -127,9 +212,26 @@ impl FromStr for NonFungibleId {
     }
 }
 
+impl fmt::Display for NonFungibleIdType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match self {
+            NonFungibleIdType::Bytes => write!(f, "Bytes"),
+            NonFungibleIdType::Number => write!(f, "Number"),
+            NonFungibleIdType::String => write!(f, "String"),
+            NonFungibleIdType::UUID => write!(f, "UUID"),
+        }
+    }
+}
+
+impl fmt::Debug for NonFungibleIdType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self)
+    }
+}
+
 impl fmt::Display for NonFungibleId {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        write!(f, "{}", hex::encode(&self.0))
+        write!(f, "{}", hex::encode(&self.value))
```

Comment:
Given non-fungible IDs are strongly typed, I think it makes more sense to design a syntx, similar to manifest, instead of hex.

Reply:
100% agree.

Reply:
New task for this: SCRY-292